{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "import subprocess\n",
    "import pandas as pd\n",
    "import sys\n",
    "\n",
    "def run_process(command, print_out=1, cmd_list=True):\n",
    "    p = subprocess.Popen(command, shell=cmd_list, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)\n",
    "    outputs = []\n",
    "    if print_out:\n",
    "        for outline in p.stdout.readlines():\n",
    "#             print(outline.strip())\n",
    "            outputs.append(outline.strip().decode(\"utf-8\"))\n",
    "        \n",
    "    retval = p.wait()\n",
    "    return outputs\n",
    "\n",
    "\n",
    "\n",
    "def merge_vcf(file_list, directory, out_path):\n",
    "    \"\"\" chrom_list = [ALL.chr1.shapeit2_integrated_v1a.GRCh38.20181129.phased.vcf.gz,\n",
    "    ALL.chr2.shapeit2_integrated_v1a.GRCh38.20181129.phased.vcf.gz] # list of files to merge from directory\n",
    "    \"\"\"\n",
    "    concat_input = ''\n",
    "    num_files = len(file_list)\n",
    "    out_fp = out_path + str(num_files) + '_merged.vcf'\n",
    "    for file in file_list:\n",
    "        concat_input = concat_input + directory + file + ' '\n",
    "        \n",
    "    cmd = 'bcftools concat ' + concat_input +' -O v --output ' + out_fp\n",
    "    print(cmd)\n",
    "    run_process(cmd)\n",
    "    return out_fp\n",
    "\n",
    "def prune_filter_vcf(fp, out_path, out_prefix):\n",
    "    \"\"\" ex: fp = '/datasets/dsc180a-wi20-public/Genome/vcf/ALL.chr22.shapeit2_integrated_v1a.GRCh38.20181129.phased.vcf.gz'\n",
    "    out_path = '../data/vcf/filtered/'\n",
    "    out_prefix = 'chr22filtered'\n",
    "    MAF = '0.1'\n",
    "    \"\"\"\n",
    "    from pathlib import Path\n",
    "    Path(out_path).mkdir(parents=True, exist_ok=True)\n",
    "    \n",
    "    filtered_vcf = out_path + 'filt_' +  out_prefix\n",
    "    \n",
    "    # plink2 filter based on maf, missing genotype, sample missingness\n",
    "    run_process('plink2 --vcf ' + fp + ' --geno 0.1 --mind 0.1 --maf 0.1 --make-bed --recode vcf -out ' + filtered_vcf)\n",
    "    \n",
    "    # plink2 filters maf and creates prune.in for pca\n",
    "    cmd = [\"plink2 --vcf \" + filtered_vcf + '.vcf' + \" --indep-pairwise 50 10 0.1 --out \" + out_path + out_prefix]\n",
    "    \n",
    "    run_process(cmd)\n",
    "    return [out_path + out_prefix + '.prune.in', filtered_vcf+ '.vcf']\n",
    "    \n",
    "    \n",
    "\n",
    "\n",
    "    \n",
    "def make_pca(fp, prune_path, pca_out_path, out_prefix):\n",
    "    \"\"\" fp = '/datasets/dsc180a-wi20-public/Genome/vcf/ALL.chr22.shapeit2_integrated_v1a.GRCh38.20181129.phased.vcf.gz'\n",
    "    prune_path = '../data/vcf/filtered/chr22filtered.prune.in'\n",
    "    \n",
    "    \"\"\"\n",
    "    from pathlib import Path\n",
    "    Path(pca_out_path).mkdir(parents=True, exist_ok=True)\n",
    "    \n",
    "#     if os.path.exists(prune_path.replace('~', str(Path.home())).replace('\\'', '')): # check if my computer\n",
    "#         print('hi')\n",
    "        \n",
    "    pca_file_name = pca_out_path + out_prefix + '_pca'\n",
    "    \n",
    "    cmd = 'plink2 --vcf ' + fp + ' --extract ' + prune_path + ' --make-bed --pca --out ' + pca_file_name\n",
    "    \n",
    "    run_process(cmd)\n",
    "    \n",
    "    return pca_file_name\n",
    "    \n",
    "def plot_from_pca(pca_file_name, population_df):\n",
    "\n",
    "    import matplotlib.pyplot as plt\n",
    "\n",
    "\n",
    "    eigvec = pd.read_table(pca_file_name + \".eigenvec\", delimiter=' ', header = None)\n",
    "    eigval = pd.read_table(pca_file_name + \".eigenval\", delimiter=' ', header = None)\n",
    "\n",
    "    eigvec_wPop = pop_df.merge(eigvec, left_on='sample', right_on=0)\n",
    "\n",
    "    to_plot = eigvec_wPop.copy()\n",
    "    to_plot[0] = to_plot[0].apply(lambda x: x[:2])\n",
    "    to_plot[1] = to_plot[1].apply(lambda x: x[:2])\n",
    "\n",
    "    # ax = to_plot.plot.scatter(x=2, y=3, label='pop',legend=False)\n",
    "\n",
    "    import colorsys\n",
    "\n",
    "    N = 26\n",
    "    # HSV_tuples = [(x*1.0/N, 0.5, 0.5) for x in range(N)]\n",
    "    # RGB_tuples = map(lambda x: colorsys.hsv_to_rgb(*x), HSV_tuples)\n",
    "\n",
    "    # colors = list(RGB_tuples)\n",
    "    from random import randint\n",
    "\n",
    "    colors=[]\n",
    "    for i in range(N):\n",
    "        colors.append('#%06X' % randint(0, 0xFFFFFF))\n",
    "\n",
    "    colors = dict(zip(list(to_plot['super_pop'].unique()), colors))\n",
    "    ###\n",
    "    _, ax = plt.subplots()\n",
    "    for key,group in to_plot.groupby('super_pop'):\n",
    "        group.plot.scatter(ax=ax, x=2, y=3, label=key, color = colors[key]);\n",
    "        ax.set_title('Principal Components 1 and 2')\n",
    "        ax.set_ylabel('PC 2')\n",
    "        ax.set_xlabel('PC 1')\n",
    "        plt.savefig('pca1_2.png')\n",
    "\n",
    "    plt.show()\n",
    "    \n",
    "    ax.legend()\n",
    "\n",
    "    _, ax = plt.subplots()\n",
    "    for key,group in to_plot.groupby('super_pop'):\n",
    "        group.plot.scatter(ax=ax, x=2, y=4, label=key, color = colors[key]);\n",
    "        ax.set_title('Principal Components 1 and 3')\n",
    "        ax.set_ylabel('PC 3')\n",
    "        ax.set_xlabel('PC 1')\n",
    "        plt.savefig('pca1_3.png')\n",
    "\n",
    "    ###\n",
    "    plt.show()\n",
    "    \n",
    "\n",
    "    ax.legend()\n",
    "    perc_var = (eigval[:16])/eigval[0].sum()*100\n",
    "\n",
    "    ax2 = perc_var.plot(kind='bar', legend=False)\n",
    "    ax2.set_title('Percent Variance Explained by PC')\n",
    "    plt.savefig('pca_var.png')\n",
    "def index_bwa(ref, name_idx):\n",
    "    ''' name_idx = 'refhg38'\n",
    "    '''\n",
    "    run_process('bwa index -p ' +name_idx+ ' -a bwtsw ' + ref)\n",
    "    print('done indexing reference')\n",
    "\n",
    "def fa_to_sam(fp1, fp2, ref, refname, outpath, index=0,header=0):\n",
    "    \"\"\" http://bioinformatics-core-shared-training.github.io/cruk-bioinf-sschool/Day1/Sequence%20Alignment_July2015_ShamithSamarajiwa.pdf\n",
    "        refname: nickname after indexing the ref file\n",
    "    \"\"\"\n",
    "    if index == 1: # set refrence index\n",
    "        index_bwa(ref, name_idx)\n",
    "    space = ' '\n",
    "    \n",
    "    # 1000g is paired ends, use sampe\n",
    "    cmd = 'bwa aln -t 4 ' + refname + ' ' + fp1 +'> ' +outpath + fp1 + \".sai\"\n",
    "    sai1 = outpath + fp1 + \".sai\"\n",
    "    \n",
    "    cmd = 'bwa aln -t 4 ' + refname + ' ' + fp1 +'> '+ outpath + fp2 + \".sai\"\n",
    "    sai2 =outpath + fp2 + \".sai\"\n",
    "    \n",
    "    cmd = 'bwa sampe '+ refname + space + sai1 + space + sai2 + space +fp1+space+fp2 +' > '+ outpath + fp1 + '.sam'\n",
    "    \n",
    "    return 1\n",
    "\n",
    "def sam_to_bam(sam, ref,header=0):\n",
    "    \"\"\"if sam has header use header=1\"\"\"\n",
    "    if header==0:\n",
    "        cmd= ['samtools view -bT '+ ref+ ' ' +sam+ ' > '+ sam +'.bam'] # when no header\n",
    "    else:\n",
    "        cmd= ['samtools view -bS '+ sam + ' > '+ sam +'.bam']\n",
    "        \n",
    "    run_process(cmd)\n",
    "    run_process('samtools sort ' + sam +'.bam ' + sam +'_sorted.bam')\n",
    "    return 1\n",
    "\n",
    "def bam_to_vcf(bam, ref):\n",
    "    \"\"\"\"needs testing\"\"\"\n",
    "    run_process('samtools faidx ' + ref)\n",
    "#     run_process('samtools mpileup -g -f ' + ref +' my-sorted1.bam my-sorted-2.bam my-sorted-n.bam > myraw.bcf')\n",
    "    \n",
    "    \n",
    "def main(targets):\n",
    "    # grab small amount of data\n",
    "    if 'test-project' in targets:\n",
    "        \n",
    "        fp = './test/chr22_test.vcf.gz'\n",
    "        \n",
    "        prunefp, filtered_vcf = prune_filter_vcf(fp, \"./data/vcf/filtered/\", fp.split('/')[-1])\n",
    "        \n",
    "        pca_out_path = './data/pca/'\n",
    "        \n",
    "        # since it's test sample, filtering will get rid of too much of the variance, so input the regular data to plot \n",
    "        # usually input filtered_vcf in place of fp\n",
    "        pca_file_name = make_pca(fp, prunefp, pca_out_path, filtered_vcf.split('/')[-1])\n",
    "        \n",
    "        print(pca_file_name)\n",
    "        \n",
    "        pop_df = pd.read_csv('./test/integrated_call_samples_v3.20130502.ALL.panel', delimiter='\\t').drop(columns=['Unnamed: 4', 'Unnamed: 5'])\n",
    "        \n",
    "        plot_from_pca(pca_file_name, pop_df)\n",
    "\n",
    "    return\n",
    "\n",
    "\n",
    "if __name__ == '__main__':\n",
    "    targets = sys.argv[1:]\n",
    "    main(targets)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "b'PLINK v1.90p 64-bit (25 Mar 2016)          https://www.cog-genomics.org/plink2'\n",
      "b'(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3'\n",
      "b'Logging to ./data/vcf/filtered/filt_chr22_test.vcf.gz.vcf.log.'\n",
      "b'Options in effect:'\n",
      "b'--geno 0.1'\n",
      "b'--maf 0.1'\n",
      "b'--make-bed'\n",
      "b'--mind 0.1'\n",
      "b'--out ./data/vcf/filtered/filt_chr22_test.vcf.gz.vcf'\n",
      "b'--recode vcf'\n",
      "b'--vcf ./test/chr22_test.vcf.gz'\n",
      "b''\n",
      "b'257652 MB RAM detected; reserving 128826 MB for main workspace.'\n",
      "b'--vcf: 1k variants complete.\\r--vcf: 2k variants complete.\\r--vcf: 3k variants complete.\\r--vcf: 4k variants complete.\\r--vcf: 5k variants complete.\\r--vcf: 6k variants complete.\\r--vcf: 7k variants complete.\\r--vcf: 8k variants complete.\\r--vcf: 9k variants complete.\\r--vcf: 10k variants complete.\\r--vcf: 11k variants complete.\\r--vcf: 12k variants complete.\\r--vcf: 13k variants complete.\\r--vcf: 14k variants complete.\\r--vcf: 15k variants complete.\\r--vcf: 16k variants complete.\\r--vcf: 17k variants complete.\\r--vcf: 18k variants complete.\\r--vcf: 19k variants complete.\\r--vcf: 20k variants complete.\\r--vcf: 21k variants complete.\\r--vcf: 22k variants complete.\\r--vcf: 23k variants complete.\\r--vcf: 24k variants complete.\\r--vcf: 25k variants complete.\\r--vcf: ./data/vcf/filtered/filt_chr22_test.vcf.gz.vcf-temporary.bed +'\n",
      "b'./data/vcf/filtered/filt_chr22_test.vcf.gz.vcf-temporary.bim +'\n",
      "b'./data/vcf/filtered/filt_chr22_test.vcf.gz.vcf-temporary.fam written.'\n",
      "b'25078 variants loaded from .bim file.'\n",
      "b'2548 people (0 males, 0 females, 2548 ambiguous) loaded from .fam.'\n",
      "b'Ambiguous sex IDs written to'\n",
      "b'./data/vcf/filtered/filt_chr22_test.vcf.gz.vcf.nosex .'\n",
      "b'0 people removed due to missing genotype data (--mind).'\n",
      "b'Using 1 thread (no multithreaded calculations invoked).'\n",
      "b'Before main variant filters, 2548 founders and 0 nonfounders present.'\n",
      "b'Calculating allele frequencies... 0%\\x08\\x081%\\x08\\x082%\\x08\\x083%\\x08\\x084%\\x08\\x085%\\x08\\x086%\\x08\\x087%\\x08\\x088%\\x08\\x089%\\x08\\x0810%\\x08\\x08\\x0811%\\x08\\x08\\x0812%\\x08\\x08\\x0813%\\x08\\x08\\x0814%\\x08\\x08\\x0815%\\x08\\x08\\x0816%\\x08\\x08\\x0817%\\x08\\x08\\x0818%\\x08\\x08\\x0819%\\x08\\x08\\x0820%\\x08\\x08\\x0821%\\x08\\x08\\x0822%\\x08\\x08\\x0823%\\x08\\x08\\x0824%\\x08\\x08\\x0825%\\x08\\x08\\x0826%\\x08\\x08\\x0827%\\x08\\x08\\x0828%\\x08\\x08\\x0829%\\x08\\x08\\x0830%\\x08\\x08\\x0831%\\x08\\x08\\x0832%\\x08\\x08\\x0833%\\x08\\x08\\x0834%\\x08\\x08\\x0835%\\x08\\x08\\x0836%\\x08\\x08\\x0837%\\x08\\x08\\x0838%\\x08\\x08\\x0839%\\x08\\x08\\x0840%\\x08\\x08\\x0841%\\x08\\x08\\x0842%\\x08\\x08\\x0843%\\x08\\x08\\x0844%\\x08\\x08\\x0845%\\x08\\x08\\x0846%\\x08\\x08\\x0847%\\x08\\x08\\x0848%\\x08\\x08\\x0849%\\x08\\x08\\x0850%\\x08\\x08\\x0851%\\x08\\x08\\x0852%\\x08\\x08\\x0853%\\x08\\x08\\x0854%\\x08\\x08\\x0855%\\x08\\x08\\x0856%\\x08\\x08\\x0857%\\x08\\x08\\x0858%\\x08\\x08\\x0859%\\x08\\x08\\x0860%\\x08\\x08\\x0861%\\x08\\x08\\x0862%\\x08\\x08\\x0863%\\x08\\x08\\x0864%\\x08\\x08\\x0865%\\x08\\x08\\x0866%\\x08\\x08\\x0867%\\x08\\x08\\x0868%\\x08\\x08\\x0869%\\x08\\x08\\x0870%\\x08\\x08\\x0871%\\x08\\x08\\x0872%\\x08\\x08\\x0873%\\x08\\x08\\x0874%\\x08\\x08\\x0875%\\x08\\x08\\x0876%\\x08\\x08\\x0877%\\x08\\x08\\x0878%\\x08\\x08\\x0879%\\x08\\x08\\x0880%\\x08\\x08\\x0881%\\x08\\x08\\x0882%\\x08\\x08\\x0883%\\x08\\x08\\x0884%\\x08\\x08\\x0885%\\x08\\x08\\x0886%\\x08\\x08\\x0887%\\x08\\x08\\x0888%\\x08\\x08\\x0889%\\x08\\x08\\x0890%\\x08\\x08\\x0891%\\x08\\x08\\x0892%\\x08\\x08\\x0893%\\x08\\x08\\x0894%\\x08\\x08\\x0895%\\x08\\x08\\x0896%\\x08\\x08\\x0897%\\x08\\x08\\x0898%\\x08\\x08\\x0899%\\x08\\x08\\x08\\x08 done.'\n",
      "b'0 variants removed due to missing genotype data (--geno).'\n",
      "b'23158 variants removed due to minor allele threshold(s)'\n",
      "b'(--maf/--max-maf/--mac/--max-mac).'\n",
      "b'1920 variants and 2548 people pass filters and QC.'\n",
      "b'Note: No phenotypes present.'\n",
      "b'--make-bed to ./data/vcf/filtered/filt_chr22_test.vcf.gz.vcf.bed +'\n",
      "b'./data/vcf/filtered/filt_chr22_test.vcf.gz.vcf.bim +'\n",
      "b'./data/vcf/filtered/filt_chr22_test.vcf.gz.vcf.fam ... 0%\\x08\\x081%\\x08\\x082%\\x08\\x083%\\x08\\x084%\\x08\\x085%\\x08\\x086%\\x08\\x087%\\x08\\x088%\\x08\\x089%\\x08\\x0810%\\x08\\x08\\x0811%\\x08\\x08\\x0812%\\x08\\x08\\x0813%\\x08\\x08\\x0814%\\x08\\x08\\x0815%\\x08\\x08\\x0816%\\x08\\x08\\x0817%\\x08\\x08\\x0818%\\x08\\x08\\x0819%\\x08\\x08\\x0820%\\x08\\x08\\x0821%\\x08\\x08\\x0822%\\x08\\x08\\x0823%\\x08\\x08\\x0824%\\x08\\x08\\x0825%\\x08\\x08\\x0826%\\x08\\x08\\x0827%\\x08\\x08\\x0828%\\x08\\x08\\x0829%\\x08\\x08\\x0830%\\x08\\x08\\x0831%\\x08\\x08\\x0832%\\x08\\x08\\x0833%\\x08\\x08\\x0834%\\x08\\x08\\x0835%\\x08\\x08\\x0836%\\x08\\x08\\x0837%\\x08\\x08\\x0838%\\x08\\x08\\x0839%\\x08\\x08\\x0840%\\x08\\x08\\x0841%\\x08\\x08\\x0842%\\x08\\x08\\x0843%\\x08\\x08\\x0844%\\x08\\x08\\x0845%\\x08\\x08\\x0846%\\x08\\x08\\x0847%\\x08\\x08\\x0848%\\x08\\x08\\x0849%\\x08\\x08\\x0850%\\x08\\x08\\x0851%\\x08\\x08\\x0852%\\x08\\x08\\x0853%\\x08\\x08\\x0854%\\x08\\x08\\x0855%\\x08\\x08\\x0856%\\x08\\x08\\x0857%\\x08\\x08\\x0858%\\x08\\x08\\x0859%\\x08\\x08\\x0860%\\x08\\x08\\x0861%\\x08\\x08\\x0862%\\x08\\x08\\x0863%\\x08\\x08\\x0864%\\x08\\x08\\x0865%\\x08\\x08\\x0866%\\x08\\x08\\x0867%\\x08\\x08\\x0868%\\x08\\x08\\x0869%\\x08\\x08\\x0870%\\x08\\x08\\x0871%\\x08\\x08\\x0872%\\x08\\x08\\x0873%\\x08\\x08\\x0874%\\x08\\x08\\x0875%\\x08\\x08\\x0876%\\x08\\x08\\x0877%\\x08\\x08\\x0878%\\x08\\x08\\x0879%\\x08\\x08\\x0880%\\x08\\x08\\x0881%\\x08\\x08\\x0882%\\x08\\x08\\x0883%\\x08\\x08\\x0884%\\x08\\x08\\x0885%\\x08\\x08\\x0886%\\x08\\x08\\x0887%\\x08\\x08\\x0888%\\x08\\x08\\x0889%\\x08\\x08\\x0890%\\x08\\x08\\x0891%\\x08\\x08\\x0892%\\x08\\x08\\x0893%\\x08\\x08\\x0894%\\x08\\x08\\x0895%\\x08\\x08\\x0896%\\x08\\x08\\x0897%\\x08\\x08\\x0898%\\x08\\x08\\x0899%\\x08\\x08\\x08done.'\n",
      "b'--recode vcf to ./data/vcf/filtered/filt_chr22_test.vcf.gz.vcf.vcf ... 0%\\x08\\x081%\\x08\\x082%\\x08\\x083%\\x08\\x084%\\x08\\x085%\\x08\\x086%\\x08\\x087%\\x08\\x088%\\x08\\x089%\\x08\\x0810%\\x08\\x08\\x0811%\\x08\\x08\\x0812%\\x08\\x08\\x0813%\\x08\\x08\\x0814%\\x08\\x08\\x0815%\\x08\\x08\\x0816%\\x08\\x08\\x0817%\\x08\\x08\\x0818%\\x08\\x08\\x0819%\\x08\\x08\\x0820%\\x08\\x08\\x0821%\\x08\\x08\\x0822%\\x08\\x08\\x0823%\\x08\\x08\\x0824%\\x08\\x08\\x0825%\\x08\\x08\\x0826%\\x08\\x08\\x0827%\\x08\\x08\\x0828%\\x08\\x08\\x0829%\\x08\\x08\\x0830%\\x08\\x08\\x0831%\\x08\\x08\\x0832%\\x08\\x08\\x0833%\\x08\\x08\\x0834%\\x08\\x08\\x0835%\\x08\\x08\\x0836%\\x08\\x08\\x0837%\\x08\\x08\\x0838%\\x08\\x08\\x0839%\\x08\\x08\\x0840%\\x08\\x08\\x0841%\\x08\\x08\\x0842%\\x08\\x08\\x0843%\\x08\\x08\\x0844%\\x08\\x08\\x0845%\\x08\\x08\\x0846%\\x08\\x08\\x0847%\\x08\\x08\\x0848%\\x08\\x08\\x0849%\\x08\\x08\\x0850%\\x08\\x08\\x0851%\\x08\\x08\\x0852%\\x08\\x08\\x0853%\\x08\\x08\\x0854%\\x08\\x08\\x0855%\\x08\\x08\\x0856%\\x08\\x08\\x0857%\\x08\\x08\\x0858%\\x08\\x08\\x0859%\\x08\\x08\\x0860%\\x08\\x08\\x0861%\\x08\\x08\\x0862%\\x08\\x08\\x0863%\\x08\\x08\\x0864%\\x08\\x08\\x0865%\\x08\\x08\\x0866%\\x08\\x08\\x0867%\\x08\\x08\\x0868%\\x08\\x08\\x0869%\\x08\\x08\\x0870%\\x08\\x08\\x0871%\\x08\\x08\\x0872%\\x08\\x08\\x0873%\\x08\\x08\\x0874%\\x08\\x08\\x0875%\\x08\\x08\\x0876%\\x08\\x08\\x0877%\\x08\\x08\\x0878%\\x08\\x08\\x0879%\\x08\\x08\\x0880%\\x08\\x08\\x0881%\\x08\\x08\\x0882%\\x08\\x08\\x0883%\\x08\\x08\\x0884%\\x08\\x08\\x0885%\\x08\\x08\\x0886%\\x08\\x08\\x0887%\\x08\\x08\\x0888%\\x08\\x08\\x0889%\\x08\\x08\\x0890%\\x08\\x08\\x0891%\\x08\\x08\\x0892%\\x08\\x08\\x0893%\\x08\\x08\\x0894%\\x08\\x08\\x0895%\\x08\\x08\\x0896%\\x08\\x08\\x0897%\\x08\\x08\\x0898%\\x08\\x08\\x0899%\\x08\\x08\\x08done.'\n",
      "b'PLINK v1.90p 64-bit (25 Mar 2016)          https://www.cog-genomics.org/plink2'\n",
      "b'(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3'\n",
      "b'Logging to ./data/vcf/filtered/chr22_test.vcf.gz.log.'\n",
      "b'Options in effect:'\n",
      "b'--indep-pairwise 50 10 0.1'\n",
      "b'--out ./data/vcf/filtered/chr22_test.vcf.gz'\n",
      "b'--vcf ./data/vcf/filtered/filt_chr22_test.vcf.gz.vcf'\n",
      "b''\n",
      "b'257652 MB RAM detected; reserving 128826 MB for main workspace.'\n",
      "b'--vcf: 1k variants complete.\\r--vcf: ./data/vcf/filtered/chr22_test.vcf.gz-temporary.bed +'\n",
      "b'./data/vcf/filtered/chr22_test.vcf.gz-temporary.bim +'\n",
      "b'./data/vcf/filtered/chr22_test.vcf.gz-temporary.fam written.'\n",
      "b'1920 variants loaded from .bim file.'\n",
      "b'2548 people (0 males, 0 females, 2548 ambiguous) loaded from .fam.'\n",
      "b'Ambiguous sex IDs written to ./data/vcf/filtered/chr22_test.vcf.gz.nosex .'\n",
      "b'Using 1 thread (no multithreaded calculations invoked).'\n",
      "b'Before main variant filters, 2548 founders and 0 nonfounders present.'\n",
      "b'Calculating allele frequencies... 0%\\x08\\x081%\\x08\\x082%\\x08\\x083%\\x08\\x084%\\x08\\x085%\\x08\\x086%\\x08\\x087%\\x08\\x088%\\x08\\x089%\\x08\\x0810%\\x08\\x08\\x0811%\\x08\\x08\\x0812%\\x08\\x08\\x0813%\\x08\\x08\\x0814%\\x08\\x08\\x0815%\\x08\\x08\\x0816%\\x08\\x08\\x0817%\\x08\\x08\\x0818%\\x08\\x08\\x0819%\\x08\\x08\\x0820%\\x08\\x08\\x0821%\\x08\\x08\\x0822%\\x08\\x08\\x0823%\\x08\\x08\\x0824%\\x08\\x08\\x0825%\\x08\\x08\\x0826%\\x08\\x08\\x0827%\\x08\\x08\\x0828%\\x08\\x08\\x0829%\\x08\\x08\\x0830%\\x08\\x08\\x0831%\\x08\\x08\\x0832%\\x08\\x08\\x0833%\\x08\\x08\\x0834%\\x08\\x08\\x0835%\\x08\\x08\\x0836%\\x08\\x08\\x0837%\\x08\\x08\\x0838%\\x08\\x08\\x0839%\\x08\\x08\\x0840%\\x08\\x08\\x0841%\\x08\\x08\\x0842%\\x08\\x08\\x0843%\\x08\\x08\\x0844%\\x08\\x08\\x0845%\\x08\\x08\\x0846%\\x08\\x08\\x0847%\\x08\\x08\\x0848%\\x08\\x08\\x0849%\\x08\\x08\\x0850%\\x08\\x08\\x0851%\\x08\\x08\\x0852%\\x08\\x08\\x0853%\\x08\\x08\\x0854%\\x08\\x08\\x0855%\\x08\\x08\\x0856%\\x08\\x08\\x0857%\\x08\\x08\\x0858%\\x08\\x08\\x0859%\\x08\\x08\\x0860%\\x08\\x08\\x0861%\\x08\\x08\\x0862%\\x08\\x08\\x0863%\\x08\\x08\\x0864%\\x08\\x08\\x0865%\\x08\\x08\\x0866%\\x08\\x08\\x0867%\\x08\\x08\\x0868%\\x08\\x08\\x0869%\\x08\\x08\\x0870%\\x08\\x08\\x0871%\\x08\\x08\\x0872%\\x08\\x08\\x0873%\\x08\\x08\\x0874%\\x08\\x08\\x0875%\\x08\\x08\\x0876%\\x08\\x08\\x0877%\\x08\\x08\\x0878%\\x08\\x08\\x0879%\\x08\\x08\\x0880%\\x08\\x08\\x0881%\\x08\\x08\\x0882%\\x08\\x08\\x0883%\\x08\\x08\\x0884%\\x08\\x08\\x0885%\\x08\\x08\\x0886%\\x08\\x08\\x0887%\\x08\\x08\\x0888%\\x08\\x08\\x0889%\\x08\\x08\\x0890%\\x08\\x08\\x0891%\\x08\\x08\\x0892%\\x08\\x08\\x0893%\\x08\\x08\\x0894%\\x08\\x08\\x0895%\\x08\\x08\\x0896%\\x08\\x08\\x0897%\\x08\\x08\\x0898%\\x08\\x08\\x0899%\\x08\\x08\\x08\\x08 done.'\n",
      "b'1920 variants and 2548 people pass filters and QC.'\n",
      "b'Note: No phenotypes present.'\n",
      "b'1%\\r2%\\r3%\\r4%\\r5%\\r6%\\r7%\\r8%\\r9%\\r10%\\r11%\\r11%\\r12%\\r13%\\r14%\\r15%\\r16%\\r17%\\r18%\\r19%\\r20%\\r21%\\r22%\\r23%\\r23%\\r24%\\r25%\\r26%\\r27%\\r28%\\r29%\\r30%\\r31%\\r32%\\r33%\\r34%\\r35%\\r36%\\r36%\\r37%\\r38%\\r39%\\r40%\\r41%\\r42%\\r43%\\r44%\\r45%\\r46%\\r47%\\r48%\\r48%\\r49%\\r50%\\r51%\\r52%\\r53%\\r54%\\r55%\\r56%\\r57%\\r58%\\r59%\\r60%\\r61%\\r61%\\r62%\\r63%\\r64%\\r65%\\r66%\\r67%\\r68%\\r69%\\r70%\\r71%\\r72%\\r73%\\r73%\\r74%\\r75%\\r76%\\r77%\\r78%\\r79%\\r80%\\r81%\\r82%\\r83%\\r84%\\r85%\\r86%\\r86%\\r87%\\r88%\\r89%\\r90%\\r91%\\r92%\\r93%\\r94%\\r95%\\r96%\\r97%\\r98%\\r98%\\r99%\\rPruned 1851 variants from chromosome 22, leaving 69.'\n",
      "b'Pruning complete.  1851 of 1920 variants removed.'\n",
      "b'Writing...\\rMarker lists written to ./data/vcf/filtered/chr22_test.vcf.gz.prune.in and'\n",
      "b'./data/vcf/filtered/chr22_test.vcf.gz.prune.out .'\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "b'PLINK v1.90p 64-bit (25 Mar 2016)          https://www.cog-genomics.org/plink2'\n",
      "b'(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3'\n",
      "b'Logging to ./data/pca/filt_chr22_test.vcf.gz.vcf_pca.log.'\n",
      "b'Options in effect:'\n",
      "b'--extract ./data/vcf/filtered/chr22_test.vcf.gz.prune.in'\n",
      "b'--make-bed'\n",
      "b'--out ./data/pca/filt_chr22_test.vcf.gz.vcf_pca'\n",
      "b'--pca'\n",
      "b'--vcf ./test/chr22_test.vcf.gz'\n",
      "b''\n",
      "b'257652 MB RAM detected; reserving 128826 MB for main workspace.'\n",
      "b'--vcf: 1k variants complete.\\r--vcf: 2k variants complete.\\r--vcf: 3k variants complete.\\r--vcf: 4k variants complete.\\r--vcf: 5k variants complete.\\r--vcf: 6k variants complete.\\r--vcf: 7k variants complete.\\r--vcf: 8k variants complete.\\r--vcf: 9k variants complete.\\r--vcf: 10k variants complete.\\r--vcf: 11k variants complete.\\r--vcf: 12k variants complete.\\r--vcf: 13k variants complete.\\r--vcf: 14k variants complete.\\r--vcf: 15k variants complete.\\r--vcf: 16k variants complete.\\r--vcf: 17k variants complete.\\r--vcf: 18k variants complete.\\r--vcf: 19k variants complete.\\r--vcf: 20k variants complete.\\r--vcf: 21k variants complete.\\r--vcf: 22k variants complete.\\r--vcf: 23k variants complete.\\r--vcf: 24k variants complete.\\r--vcf: 25k variants complete.\\r--vcf: ./data/pca/filt_chr22_test.vcf.gz.vcf_pca-temporary.bed +'\n",
      "b'./data/pca/filt_chr22_test.vcf.gz.vcf_pca-temporary.bim +'\n",
      "b'./data/pca/filt_chr22_test.vcf.gz.vcf_pca-temporary.fam written.'\n",
      "b'25078 variants loaded from .bim file.'\n",
      "b'2548 people (0 males, 0 females, 2548 ambiguous) loaded from .fam.'\n",
      "b'Ambiguous sex IDs written to ./data/pca/filt_chr22_test.vcf.gz.vcf_pca.nosex .'\n",
      "b'--extract: 25078 variants remaining.'\n",
      "b'Warning: At least 68 duplicate IDs in --extract file.'\n",
      "b'Using up to 127 threads (change this with --threads).'\n",
      "b'Before main variant filters, 2548 founders and 0 nonfounders present.'\n",
      "b'Calculating allele frequencies... 0%\\x08\\x081%\\x08\\x082%\\x08\\x083%\\x08\\x084%\\x08\\x085%\\x08\\x086%\\x08\\x087%\\x08\\x088%\\x08\\x089%\\x08\\x0810%\\x08\\x08\\x0811%\\x08\\x08\\x0812%\\x08\\x08\\x0813%\\x08\\x08\\x0814%\\x08\\x08\\x0815%\\x08\\x08\\x0816%\\x08\\x08\\x0817%\\x08\\x08\\x0818%\\x08\\x08\\x0819%\\x08\\x08\\x0820%\\x08\\x08\\x0821%\\x08\\x08\\x0822%\\x08\\x08\\x0823%\\x08\\x08\\x0824%\\x08\\x08\\x0825%\\x08\\x08\\x0826%\\x08\\x08\\x0827%\\x08\\x08\\x0828%\\x08\\x08\\x0829%\\x08\\x08\\x0830%\\x08\\x08\\x0831%\\x08\\x08\\x0832%\\x08\\x08\\x0833%\\x08\\x08\\x0834%\\x08\\x08\\x0835%\\x08\\x08\\x0836%\\x08\\x08\\x0837%\\x08\\x08\\x0838%\\x08\\x08\\x0839%\\x08\\x08\\x0840%\\x08\\x08\\x0841%\\x08\\x08\\x0842%\\x08\\x08\\x0843%\\x08\\x08\\x0844%\\x08\\x08\\x0845%\\x08\\x08\\x0846%\\x08\\x08\\x0847%\\x08\\x08\\x0848%\\x08\\x08\\x0849%\\x08\\x08\\x0850%\\x08\\x08\\x0851%\\x08\\x08\\x0852%\\x08\\x08\\x0853%\\x08\\x08\\x0854%\\x08\\x08\\x0855%\\x08\\x08\\x0856%\\x08\\x08\\x0857%\\x08\\x08\\x0858%\\x08\\x08\\x0859%\\x08\\x08\\x0860%\\x08\\x08\\x0861%\\x08\\x08\\x0862%\\x08\\x08\\x0863%\\x08\\x08\\x0864%\\x08\\x08\\x0865%\\x08\\x08\\x0866%\\x08\\x08\\x0867%\\x08\\x08\\x0868%\\x08\\x08\\x0869%\\x08\\x08\\x0870%\\x08\\x08\\x0871%\\x08\\x08\\x0872%\\x08\\x08\\x0873%\\x08\\x08\\x0874%\\x08\\x08\\x0875%\\x08\\x08\\x0876%\\x08\\x08\\x0877%\\x08\\x08\\x0878%\\x08\\x08\\x0879%\\x08\\x08\\x0880%\\x08\\x08\\x0881%\\x08\\x08\\x0882%\\x08\\x08\\x0883%\\x08\\x08\\x0884%\\x08\\x08\\x0885%\\x08\\x08\\x0886%\\x08\\x08\\x0887%\\x08\\x08\\x0888%\\x08\\x08\\x0889%\\x08\\x08\\x0890%\\x08\\x08\\x0891%\\x08\\x08\\x0892%\\x08\\x08\\x0893%\\x08\\x08\\x0894%\\x08\\x08\\x0895%\\x08\\x08\\x0896%\\x08\\x08\\x0897%\\x08\\x08\\x0898%\\x08\\x08\\x0899%\\x08\\x08\\x08\\x08 done.'\n",
      "b'25078 variants and 2548 people pass filters and QC.'\n",
      "b'Note: No phenotypes present.'\n",
      "b'60 markers complete.\\r120 markers complete.\\r180 markers complete.\\r240 markers complete.\\r300 markers complete.\\r360 markers complete.\\r420 markers complete.\\r480 markers complete.\\r540 markers complete.\\r600 markers complete.\\r660 markers complete.\\r720 markers complete.\\r780 markers complete.\\r840 markers complete.\\r900 markers complete.\\r960 markers complete.\\r1020 markers complete.\\r1080 markers complete.\\r1140 markers complete.\\r1200 markers complete.\\r1260 markers complete.\\r1320 markers complete.\\r1380 markers complete.\\r1440 markers complete.\\r1500 markers complete.\\r1560 markers complete.\\r1620 markers complete.\\r1680 markers complete.\\r1740 markers complete.\\r1800 markers complete.\\r1860 markers complete.\\r1920 markers complete.\\r1980 markers complete.\\r2040 markers complete.\\r2100 markers complete.\\r2160 markers complete.\\r2220 markers complete.\\r2280 markers complete.\\r2340 markers complete.\\r2400 markers complete.\\r2460 markers complete.\\r2520 markers complete.\\r2580 markers complete.\\r2640 markers complete.\\r2700 markers complete.\\r2760 markers complete.\\r2820 markers complete.\\r2880 markers complete.\\r2940 markers complete.\\r3000 markers complete.\\r3060 markers complete.\\r3120 markers complete.\\r3180 markers complete.\\r3240 markers complete.\\r3300 markers complete.\\r3360 markers complete.\\r3420 markers complete.\\r3480 markers complete.\\r3540 markers complete.\\r3600 markers complete.\\r3660 markers complete.\\r3720 markers complete.\\r3780 markers complete.\\r3840 markers complete.\\r3900 markers complete.\\r3960 markers complete.\\r4020 markers complete.\\r4080 markers complete.\\r4140 markers complete.\\r4200 markers complete.\\r4260 markers complete.\\r4320 markers complete.\\r4380 markers complete.\\r4440 markers complete.\\r4500 markers complete.\\r4560 markers complete.\\r4620 markers complete.\\r4680 markers complete.\\r4740 markers complete.\\r4800 markers complete.\\r4860 markers complete.\\r4920 markers complete.\\r4980 markers complete.\\r5040 markers complete.\\r5100 markers complete.\\r5160 markers complete.\\r5220 markers complete.\\r5280 markers complete.\\r5340 markers complete.\\r5400 markers complete.\\r5460 markers complete.\\r5520 markers complete.\\r5580 markers complete.\\r5640 markers complete.\\r5700 markers complete.\\r5760 markers complete.\\r5820 markers complete.\\r5880 markers complete.\\r5940 markers complete.\\r6000 markers complete.\\r6060 markers complete.\\r6120 markers complete.\\r6180 markers complete.\\r6240 markers complete.\\r6300 markers complete.\\r6360 markers complete.\\r6420 markers complete.\\r6480 markers complete.\\r6540 markers complete.\\r6600 markers complete.\\r6660 markers complete.\\r6720 markers complete.\\r6780 markers complete.\\r6840 markers complete.\\r6900 markers complete.\\r6960 markers complete.\\r7020 markers complete.\\r7080 markers complete.\\r7140 markers complete.\\r7200 markers complete.\\r7260 markers complete.\\r7320 markers complete.\\r7380 markers complete.\\r7440 markers complete.\\r7500 markers complete.\\r7560 markers complete.\\r7620 markers complete.\\r7680 markers complete.\\r7740 markers complete.\\r7800 markers complete.\\r7860 markers complete.\\r7920 markers complete.\\r7980 markers complete.\\r8040 markers complete.\\r8100 markers complete.\\r8160 markers complete.\\r8220 markers complete.\\r8280 markers complete.\\r8340 markers complete.\\r8400 markers complete.\\r8460 markers complete.\\r8520 markers complete.\\r8580 markers complete.\\r8640 markers complete.\\r8700 markers complete.\\r8760 markers complete.\\r8820 markers complete.\\r8880 markers complete.\\r8940 markers complete.\\r9000 markers complete.\\r9060 markers complete.\\r9120 markers complete.\\r9180 markers complete.\\r9240 markers complete.\\r9300 markers complete.\\r9360 markers complete.\\r9420 markers complete.\\r9480 markers complete.\\r9540 markers complete.\\r9600 markers complete.\\r9660 markers complete.\\r9720 markers complete.\\r9780 markers complete.\\r9840 markers complete.\\r9900 markers complete.\\r9960 markers complete.\\r10020 markers complete.\\r10080 markers complete.\\r10140 markers complete.\\r10200 markers complete.\\r10260 markers complete.\\r10320 markers complete.\\r10380 markers complete.\\r10440 markers complete.\\r10500 markers complete.\\r10560 markers complete.\\r10620 markers complete.\\r10680 markers complete.\\r10740 markers complete.\\r10800 markers complete.\\r10860 markers complete.\\r10920 markers complete.\\r10980 markers complete.\\r11040 markers complete.\\r11100 markers complete.\\r11160 markers complete.\\r11220 markers complete.\\r11280 markers complete.\\r11340 markers complete.\\r11400 markers complete.\\r11460 markers complete.\\r11520 markers complete.\\r11580 markers complete.\\r11640 markers complete.\\r11700 markers complete.\\r11760 markers complete.\\r11820 markers complete.\\r11880 markers complete.\\r11940 markers complete.\\r12000 markers complete.\\r12060 markers complete.\\r12120 markers complete.\\r12180 markers complete.\\r12240 markers complete.\\r12300 markers complete.\\r12360 markers complete.\\r12420 markers complete.\\r12480 markers complete.\\r12540 markers complete.\\r12600 markers complete.\\r12660 markers complete.\\r12720 markers complete.\\r12780 markers complete.\\r12840 markers complete.\\r12900 markers complete.\\r12960 markers complete.\\r13020 markers complete.\\r13080 markers complete.\\r13140 markers complete.\\r13200 markers complete.\\r13260 markers complete.\\r13320 markers complete.\\r13380 markers complete.\\r13440 markers complete.\\r13500 markers complete.\\r13560 markers complete.\\r13620 markers complete.\\r13680 markers complete.\\r13740 markers complete.\\r13800 markers complete.\\r13860 markers complete.\\r13920 markers complete.\\r13980 markers complete.\\r14040 markers complete.\\r14100 markers complete.\\r14160 markers complete.\\r14220 markers complete.\\r14280 markers complete.\\r14340 markers complete.\\r14400 markers complete.\\r14460 markers complete.\\r14520 markers complete.\\r14580 markers complete.\\r14640 markers complete.\\r14700 markers complete.\\r14760 markers complete.\\r14820 markers complete.\\r14880 markers complete.\\r14940 markers complete.\\r15000 markers complete.\\r15060 markers complete.\\r15120 markers complete.\\r15180 markers complete.\\r15240 markers complete.\\r15300 markers complete.\\r15360 markers complete.\\r15420 markers complete.\\r15480 markers complete.\\r15540 markers complete.\\r15600 markers complete.\\r15660 markers complete.\\r15720 markers complete.\\r15780 markers complete.\\r15840 markers complete.\\r15900 markers complete.\\r15960 markers complete.\\r16020 markers complete.\\r16080 markers complete.\\r16140 markers complete.\\r16200 markers complete.\\r16260 markers complete.\\r16320 markers complete.\\r16380 markers complete.\\r16440 markers complete.\\r16500 markers complete.\\r16560 markers complete.\\r16620 markers complete.\\r16680 markers complete.\\r16740 markers complete.\\r16800 markers complete.\\r16860 markers complete.\\r16920 markers complete.\\r16980 markers complete.\\r17040 markers complete.\\r17100 markers complete.\\r17160 markers complete.\\r17220 markers complete.\\r17280 markers complete.\\r17340 markers complete.\\r17400 markers complete.\\r17460 markers complete.\\r17520 markers complete.\\r17580 markers complete.\\r17640 markers complete.\\r17700 markers complete.\\r17760 markers complete.\\r17820 markers complete.\\r17880 markers complete.\\r17940 markers complete.\\r18000 markers complete.\\r18060 markers complete.\\r18120 markers complete.\\r18180 markers complete.\\r18240 markers complete.\\r18300 markers complete.\\r18360 markers complete.\\r18420 markers complete.\\r18480 markers complete.\\r18540 markers complete.\\r18600 markers complete.\\r18660 markers complete.\\r18720 markers complete.\\r18780 markers complete.\\r18840 markers complete.\\r18900 markers complete.\\r18960 markers complete.\\r19020 markers complete.\\r19080 markers complete.\\r19140 markers complete.\\r19200 markers complete.\\r19260 markers complete.\\r19320 markers complete.\\r19380 markers complete.\\r19440 markers complete.\\r19500 markers complete.\\r19560 markers complete.\\r19620 markers complete.\\r19680 markers complete.\\r19740 markers complete.\\r19800 markers complete.\\r19860 markers complete.\\r19920 markers complete.\\r19980 markers complete.\\r20040 markers complete.\\r20100 markers complete.\\r20160 markers complete.\\r20220 markers complete.\\r20280 markers complete.\\r20340 markers complete.\\r20400 markers complete.\\r20460 markers complete.\\r20520 markers complete.\\r20580 markers complete.\\r20640 markers complete.\\r20700 markers complete.\\r20760 markers complete.\\r20820 markers complete.\\r20880 markers complete.\\r20940 markers complete.\\r21000 markers complete.\\r21060 markers complete.\\r21120 markers complete.\\r21180 markers complete.\\r21240 markers complete.\\r21300 markers complete.\\r21360 markers complete.\\r21420 markers complete.\\r21480 markers complete.\\r21540 markers complete.\\r21600 markers complete.\\r21660 markers complete.\\r21720 markers complete.\\r21780 markers complete.\\r21840 markers complete.\\r21900 markers complete.\\r21960 markers complete.\\r22020 markers complete.\\r22080 markers complete.\\r22140 markers complete.\\r22200 markers complete.\\r22260 markers complete.\\r22320 markers complete.\\r22380 markers complete.\\r22440 markers complete.\\r22500 markers complete.\\r22560 markers complete.\\r22620 markers complete.\\r22680 markers complete.\\r22740 markers complete.\\r22800 markers complete.\\r22860 markers complete.\\r22920 markers complete.\\r22980 markers complete.\\r23040 markers complete.\\r23100 markers complete.\\r23160 markers complete.\\r23220 markers complete.\\r23280 markers complete.\\r23340 markers complete.\\r23400 markers complete.\\r23460 markers complete.\\r23520 markers complete.\\r23580 markers complete.\\r23640 markers complete.\\r23700 markers complete.\\r23760 markers complete.\\r23820 markers complete.\\r23880 markers complete.\\r23940 markers complete.\\r24000 markers complete.\\r24060 markers complete.\\r24120 markers complete.\\r24180 markers complete.\\r24240 markers complete.\\r24300 markers complete.\\r24360 markers complete.\\r24420 markers complete.\\r24480 markers complete.\\r24540 markers complete.\\r24600 markers complete.\\r24660 markers complete.\\r24720 markers complete.\\r24780 markers complete.\\r24840 markers complete.\\r24900 markers complete.\\r24960 markers complete.\\r25020 markers complete.\\r25078 markers complete.\\rRelationship matrix calculation complete.'\n",
      "b'[extracting eigenvalues and eigenvectors]\\r--pca: Results saved to ./data/pca/filt_chr22_test.vcf.gz.vcf_pca.eigenval and'\n",
      "b'./data/pca/filt_chr22_test.vcf.gz.vcf_pca.eigenvec .'\n",
      "b'--make-bed to ./data/pca/filt_chr22_test.vcf.gz.vcf_pca.bed +'\n",
      "b'./data/pca/filt_chr22_test.vcf.gz.vcf_pca.bim +'\n",
      "b'./data/pca/filt_chr22_test.vcf.gz.vcf_pca.fam ... 0%\\x08\\x081%\\x08\\x082%\\x08\\x083%\\x08\\x084%\\x08\\x085%\\x08\\x086%\\x08\\x087%\\x08\\x088%\\x08\\x089%\\x08\\x0810%\\x08\\x08\\x0811%\\x08\\x08\\x0812%\\x08\\x08\\x0813%\\x08\\x08\\x0814%\\x08\\x08\\x0815%\\x08\\x08\\x0816%\\x08\\x08\\x0817%\\x08\\x08\\x0818%\\x08\\x08\\x0819%\\x08\\x08\\x0820%\\x08\\x08\\x0821%\\x08\\x08\\x0822%\\x08\\x08\\x0823%\\x08\\x08\\x0824%\\x08\\x08\\x0825%\\x08\\x08\\x0826%\\x08\\x08\\x0827%\\x08\\x08\\x0828%\\x08\\x08\\x0829%\\x08\\x08\\x0830%\\x08\\x08\\x0831%\\x08\\x08\\x0832%\\x08\\x08\\x0833%\\x08\\x08\\x0834%\\x08\\x08\\x0835%\\x08\\x08\\x0836%\\x08\\x08\\x0837%\\x08\\x08\\x0838%\\x08\\x08\\x0839%\\x08\\x08\\x0840%\\x08\\x08\\x0841%\\x08\\x08\\x0842%\\x08\\x08\\x0843%\\x08\\x08\\x0844%\\x08\\x08\\x0845%\\x08\\x08\\x0846%\\x08\\x08\\x0847%\\x08\\x08\\x0848%\\x08\\x08\\x0849%\\x08\\x08\\x0850%\\x08\\x08\\x0851%\\x08\\x08\\x0852%\\x08\\x08\\x0853%\\x08\\x08\\x0854%\\x08\\x08\\x0855%\\x08\\x08\\x0856%\\x08\\x08\\x0857%\\x08\\x08\\x0858%\\x08\\x08\\x0859%\\x08\\x08\\x0860%\\x08\\x08\\x0861%\\x08\\x08\\x0862%\\x08\\x08\\x0863%\\x08\\x08\\x0864%\\x08\\x08\\x0865%\\x08\\x08\\x0866%\\x08\\x08\\x0867%\\x08\\x08\\x0868%\\x08\\x08\\x0869%\\x08\\x08\\x0870%\\x08\\x08\\x0871%\\x08\\x08\\x0872%\\x08\\x08\\x0873%\\x08\\x08\\x0874%\\x08\\x08\\x0875%\\x08\\x08\\x0876%\\x08\\x08\\x0877%\\x08\\x08\\x0878%\\x08\\x08\\x0879%\\x08\\x08\\x0880%\\x08\\x08\\x0881%\\x08\\x08\\x0882%\\x08\\x08\\x0883%\\x08\\x08\\x0884%\\x08\\x08\\x0885%\\x08\\x08\\x0886%\\x08\\x08\\x0887%\\x08\\x08\\x0888%\\x08\\x08\\x0889%\\x08\\x08\\x0890%\\x08\\x08\\x0891%\\x08\\x08\\x0892%\\x08\\x08\\x0893%\\x08\\x08\\x0894%\\x08\\x08\\x0895%\\x08\\x08\\x0896%\\x08\\x08\\x0897%\\x08\\x08\\x0898%\\x08\\x08\\x0899%\\x08\\x08\\x08done.'\n",
      "./data/pca/filt_chr22_test.vcf.gz.vcf_pca\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "main('test-project')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
